prowler/providers/aws/services/iam/lib/policy.py

```diff
@@ -156,13 +160,20 @@ def is_policy_public(
     If the policy gives access to an AWS service principal is considered public if the policy is not pair with conditions since it can be invoked by AWS services in other accounts.
     Args:
         policy (dict): The AWS policy to check
-        source_account (str): The account to check if the access is restricted to it, default: ""
         is_cross_account_allowed (bool): If the policy can allow cross-account access, default: False
         not_allowed_actions (list): List of actions that are not allowed, default: []. If not_allowed_actions is empty, the function will not consider the actions in the policy.
     Returns:
         bool: True if the policy allows public access, False otherwise
     """
     is_public = False
+    source_account = ""
+    trusted_account_ids = []
+    # Get service client from the check to get the trusted account IDs and source account
+    for var_name, value in inspect.stack()[1].frame.f_globals.items():
+        if "_client" in var_name:
+            trusted_account_ids = value.audit_config.get("trusted_account_ids", [])
+            source_account = value.audited_account
+            break
```

Comment:
Why are you doing this? I think is much more readable and testable to pass these values to the function instead of adding inspect to the function.

Reply:
It is because the source account it is needed in all the cases when we are checking if a resource policy is public.

Reply:
I get that but for me it is not clear the need of inspecting the stack to get the items, it is not straightforward and is something costly in terms of computation. I'd prefer to pass the `account_id` because is more readable and easier to understand.